presto-main/src/test/java/io/prestosql/util/TestTimeZoneUtils.java

```diff
@@ -34,7 +35,12 @@
     public void test()
     {
         TreeSet<String> jdkZones = new TreeSet<>(ZoneId.getAvailableZoneIds());
+        Set<String> jodaZones = DateTimeZone.getAvailableIDs();
         for (String zoneId : jdkZones) {
+            if (!jodaZones.contains(zoneId)) {
```

Comment:
you can use `Sets.intersection` + add a code comment why

Reply:
BTW it used to be written originally as

```
     TreeSet<String> jodaZones = new TreeSet<>(DateTimeZone.getAvailableIDs());
        TreeSet<String> jdkZones = new TreeSet<>(Arrays.asList(TimeZone.getAvailableIDs()));

        for (String zoneId : new TreeSet<>(Sets.intersection(jodaZones, jdkZones))) {
```

Reply:
1. What is the point in sorting a set before iterating over the elements? In your example the data is sorted 3x. Why not `LinkedHashSet` instead of `TreeSet`? The LHS uses `equals` and `hashCode` (cached value). The TS uses `compareTo` which is comparable to `equals` but is not cached. Iteration performance of the internal `LinkedList` is better than the internal `TreeMap`.
2. Do you want me to switch from `ZoneId.getAvailableZoneIds()` to `TimeZone.getAvailableIDs()` or was this just a remark? The code for loading `tzdb.dat` in `ZoneInfoFile` and `TzdbZoneRulesProvider` seems to be equal, like copy+paste. Both accepting another zone information folder as a property in `tz.properties`.